react-game/react-game/src/Components/AppContent.js

```diff
@@ -0,0 +1,136 @@
+
+ constructor(props) {
+  super(props);
+  this.state = JSON.parse(localStorage.getItem('state')) ? JSON.parse(localStorage.getItem('state')) : {
+    cards: [21,22,23,24,31,32,33,34,41,42,43,44,51,52,53,54,61,62,63,64,71,72,73,74,81,82,83,84,91,92,93,94,101,102,103,104,'j1', 'j2', 'j3', 'j4', 'q1','q2','q3','q4','k1','k2','k3','k4','a1','a2','a3','a4']
+  ,score: 0, timer: 300, stoptimer: false, newgame: false, gameover: false, scoreList:[], volume:40};
+  this.timers = this.state.timer < 300 ? setInterval(()=> {
+    this.setState({timer: this.state.timer - 1});
+    localStorage.setItem('state',JSON.stringify(this.state));
+  if (this.state.timer === 0 || this.state.stoptimer) {
+    clearInterval(this.timers)}
+    if (this.state.timer === 0) {this.gameOver.call(this);}
+  }, 1000) : null;
+  this.setState({cards: this.state.cards.sort(() => Math.random() - 0.5)});
+}
+
+deletecard() {
+  let arr = this.state.cards;
+  let a = arr.pop();
+  this.setState({cards: arr});
+  localStorage.setItem('state',JSON.stringify(this.state));
+  if (this.state.cards.length === 0) {this.gameOver.call(this);}
+ return a;
+}
+
+sumScore(e) {
+this.setState({score: this.state.score += e});
+localStorage.setItem('state',JSON.stringify(this.state));
+}
+
+stoptimer() {
+  this.setState({stoptimer: true});
+  localStorage.setItem('state',JSON.stringify(this.state));
+  return this.state.timer;
+}
+
+timer() {
+    this.timers = setInterval(()=> {
+      this.setState({timer: this.state.timer - 1});
+      localStorage.setItem('state',JSON.stringify(this.state));
+    if (this.state.timer === 0 || this.state.stoptimer) {
+      clearInterval(this.timers)}
+      if (this.state.timer === 0) {this.gameOver.call(this);}
+    }, 1000);
+}
+
+newGame() {
+  let cardsArr = [21,22,23,24,31,32,33,34,41,42,43,44,51,52,53,54,61,62,63,64,71,72,73,74,81,82,83,84,91,92,93,94,101,102,103,104,'j1', 'j2', 'j3', 'j4', 'q1','q2','q3','q4','k1','k2','k3','k4','a1','a2','a3','a4'];
+  cardsArr.sort(() => Math.random() - 0.5);
+  this.setState({ cards: cardsArr,stoptimer: true, score: 0});
+  this.setState({newgame: true});
+  setTimeout(() => {
+    this.setState({ timer: 300, newgame: false, stoptimer: false });
+    localStorage.setItem('state',JSON.stringify(this.state));
+  }, 1000);
+}
+
+gameOver() {
+  this.setState({gameover: true});
+  localStorage.setItem('state',JSON.stringify(this.state));
+}
+
+pushScore(name) {
+  let arr = this.state.scoreList;
+
+  const date = new Date();
+  const today = `${date.getDay() < 10 ? `0${date.getDay()}` : date.getDay()}.${date.getMonth() + 1 < 10 ? `0${date.getMonth() + 1}` : date.getMonth() + 1}.${date.getFullYear()} в ${date.getHours()}:${date.getMinutes()}`;
+
+  let element = [name, this.state.score, today];
+  if (arr.length === 10) {
+    arr =  [element, ...arr.slice(1)]
+  } else {
+    arr = [element, ...arr];
+  }
+  this.setState({scoreList: arr, gameOver: false});
+  this.setState({gameover: false});

```

Comment:
Instead of
this.setState({scoreList: arr, gameOver: false});
this.setState({gameover: false});

-> this.setState({scoreList: arr, gameOver: false, gameover: false});

What the differences between gameOver and gameover